qiskit_nature/circuit/library/ansatzes/chc.py

```diff
@@ -0,0 +1,312 @@
+
+        if initial_state is not None:
+            self._initial_state = initial_state
+
+    @property
+    def num_qubits(self) -> int:
+        """Number of qubits of the variational form.
+
+        Returns:
+           int:  An integer indicating the number of qubits.
+        """
+        return self._num_qubits
+
+    @num_qubits.setter
+    def num_qubits(self, num_qubits: int) -> None:
+        """Set the number of qubits of the variational form.
+
+        Args:
+           num_qubits: An integer indicating the number of qubits.
+        """
+        if self._num_qubits != num_qubits:
+            # invalidate the circuit
+            self._invalidate()
+            self._num_qubits = num_qubits
+            self.qregs = [QuantumRegister(num_qubits, name='q')]
+
+    @property
+    def excitations(self) -> List[Tuple[Tuple[Any, ...], ...]]:
+        """The excitation indices to be included in the circuit."""
+        return self._excitations
+
+    @excitations.setter
+    def excitations(self, excitations: List[Tuple[Tuple[Any, ...], ...]]) -> None:
+        """Sets the excitation indices to be included in the circuit."""
+        self._invalidate()
+        self._excitations = excitations
+        self._num_parameters = len(excitations) * self._reps
+        self._bounds = [(-np.pi, np.pi)] * self._num_parameters
+
+    @property
+    def initial_state(self) -> QuantumCircuit:
+        """The initial state."""
+        return self._initial_state
+
+    @initial_state.setter
+    def initial_state(self, initial_state: QuantumCircuit) -> None:
+        """Sets the initial state."""
+        self._invalidate()
+        self._initial_state = initial_state
+
+    def _check_configuration(self, raise_on_failure: bool = True) -> bool:
+        """Check if the configuration of the CHC class is valid.
+
+        Args:
+            raise_on_failure: Whether to raise on failure.
+
+        Returns:
+            True, if the configuration is valid and the circuit can be constructed. Otherwise
+            an ValueError is raised.
+
+        Raises:
+            ValueError: If the number of qubits is not specified.
+            ValueError: If the number of parameters is not specified.
+            ValueError: If the excitation list is not specified.
+        """
+        error_msg = 'The %s is None but must be set before the circuit can be built.'
+        if self._num_qubits is None:
+            if raise_on_failure:
+                raise ValueError(error_msg, 'number of qubits')
+            return False
+
+        if self._num_parameters is None:
+            if raise_on_failure:
+                raise ValueError(error_msg, 'number of parameters')
+            return False
+
+        if self._excitations is None:
+            if raise_on_failure:
+                raise ValueError(error_msg, 'excitation list')
+            return False
+
+        return True
+
+    @property
+    def ordered_parameters(self) -> List[Parameter]:
```

Comment:
I don't see `ordered_parameters` being deprecated in `NLocal`.. But nonetheless, I also found it rather awkward. Since we really only use it during `_build()`, I have replaced it with a `ParameterVector` instead.